src/Http/Fixture.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Sammyjo20\Saloon\Http;
+
+use Sammyjo20\Saloon\Helpers\Storage;
+use Sammyjo20\Saloon\Data\FixtureData;
+use Sammyjo20\Saloon\Helpers\MockConfig;
+use Sammyjo20\Saloon\Exceptions\FixtureMissingException;
+use Sammyjo20\Saloon\Exceptions\DirectoryNotFoundException;
+
+class Fixture
+{
+    /**
+     * The extension used by the fixture
+     *
+     * @var string
+     */
+    public static string $fixtureExtension = 'json';
+
+    /**
+     * The name of the fixture
+     *
+     * @var string
+     */
+    protected string $name;
+
+    /**
+     * The storage helper
+     *
+     * @var Storage
+     */
+    protected Storage $storage;
+
+    /**
+     * Constructor
+     *
+     * @param string $name
+     * @param Storage|null $storage
+     * @throws DirectoryNotFoundException
+     */
+    public function __construct(string $name, Storage $storage = null)
+    {
+        $this->name = $name;
+        $this->storage = $storage ?? new Storage(MockConfig::getFixturePath());
+    }
+
+    /**
+     * Attempt to get the mock response from the fixture.
+     *
+     * @return MockResponse|null
+     * @throws FixtureMissingException
+     * @throws \JsonException
+     */
+    public function getMockResponse(): ?MockResponse
```

Comment:
How about making that method request aware!? This would mean that the request could be passed to `getFixturePath()` and would lead to the option that `$this->name` could be a `Closure` that returns the fixture name `string` based on the current request being handled.

With that in place everyone would be totally free how to name the fixtures.

Reply:
Thanks for the review @Gummibeer 

My only concern with guessing the fixture names is that it could record unexpected results. For example, let's say I wrote a fixture like this:

```php
Saloon::fake([
    UserRequest => MockResponse::fixture(),
]);
```

That would be fairly easy to guess because the fixture name could just be the class name or the class's file path, similar to what Spatie does in their snapshot assertions package. 

My concern is when something like this happens

```php
Saloon::fake([
    MockResponse::fixture(),
]);

// or

Saloon::fake([
    '*' => MockResponse::fixture(),
]);
```

Since the fixture doesn't have name, if we were to make it request aware, it would pick up the very first request and store the fixture data against that request without the user really knowing that it was recording that request. If someone then passes a different request in, it will record that one separately as well or if they happened to have the same request, it would return that request's fixture.

I think the issue is that from a readability perspective just `MockResponse::fixture()` doesn't say anything about the request, however if it's something like this

```php
Saloon::fake([
    MockResponse::fixture('userMock'),
]);
```

It becomes similar to PHP-VCR's "cassette" functionality where the user explicitly requests a given mock. 

By the way, you can still use a closure if you would like to auto-generate the name for you, for example

```php
Saloon::fake([
    '*' => function (SaloonRequest $request) {
           return MockResponse::fixture($request::class); // Provide any name here
    }
]);
```

Reply:
I have provided a better example where you can define your own name generator for fixtures. I think this is what you are after? :D 

```php
<?php

$mockClient = new MockClient([
    '*' => function (SaloonRequest $request) {
        $reflection = new ReflectionClass($reequest);

        return MockResponse::fixture($reflection->getShortName());
    },
]);

// This will create a fixture called "GetForgeServerRequest"

$request = new GetForgeServerRequest(12345);
$response = $request->send($mockClient);

// This will create a fixture called "GetAllForgeServersRequest"

$request = new GetAllForgeServersRequest($data);
$response = $request->send($mockClient);
```

Reply:
Regarding guessing and spatie: if you check the link on depth:
It uses the test class and method name plus an Index. So every request execution will have its own name based on the test it was executed in.

Regarding the anonymous function: yes, that's exactly it. Just had it switched in order in mind.

```php
Mock response::fixture(fn(Request $request) => ...)
```

But result is the same so totally fine for me! 🥳